duckdb-vortex/src/include/rust_vector_buffer.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include <duckdb.h>
+#include <duckdb/common/types/vector_buffer.hpp>
+
+extern "C" {
+
+typedef struct FFIDuckDBBuffer FFIDuckDBBuffer;
+
+void FFIDuckDBBuffer_free(FFIDuckDBBuffer *);
+}
+
+class RustVectorBuffer : public duckdb::VectorBuffer {
+public:
+	RustVectorBuffer(FFIDuckDBBuffer *wrapper) : wrapper(wrapper) {};
+
+	~RustVectorBuffer() {
```

Comment:
`~RustVectorBuffer() override` to compile time check and be explicit that the superclass has a virtual dtor.